scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -542,19 +546,36 @@ class ScioContext private[scio] (val options: PipelineOptions,
     }
   }
 
-  /**
-   * Get an SCollection for a BigQuery table.
-   * @group input
-   */
-  def bigQueryTable(table: TableReference): SCollection[TableRow] = requireNotClosed {
+  private def bqReadTable[T: ClassTag](typedRead: bqio.BigQueryIO.TypedRead[T],
+                                         table: TableReference)
+  : SCollection[T] = requireNotClosed {
     val tableSpec: String = bqio.BigQueryHelpers.toTableSpec(table)
     if (this.isTest) {
-      this.getTestInput(BigQueryIO(tableSpec))
+      this.getTestInput(BigQueryIO[T](tableSpec))
     } else {
-      wrap(this.applyInternal(bqio.BigQueryIO.readTableRows().from(table))).setName(tableSpec)
+      wrap(this.applyInternal(typedRead.from(table))).setName(tableSpec)
     }
   }
 
+  /**
+   * Get an SCollection for a BigQuery SELECT query.
+   * Both [[https://cloud.google.com/bigquery/docs/reference/legacy-sql Legacy SQL]] and
+   * [[https://cloud.google.com/bigquery/docs/reference/standard-sql/ Standard SQL]] dialects are
+   * supported. By default the query dialect will be automatically detected. To override this
+   * behavior, start the query string with `#legacysql` or `#standardsql`.
+   * @group input
+   */
+  def bigQuerySelect(sqlQuery: String,
+                     flattenResults: Boolean = false): SCollection[TableRow] =
+    bqReadQuery(bqio.BigQueryIO.readTableRows(), sqlQuery, flattenResults)
+
+  /**
+   * Get an SCollection for a BigQuery table.
+   * @group input
+   */
+  def bigQueryTable(table: TableReference): SCollection[TableRow] =
+    bqReadTable(bqio.BigQueryIO.readTableRows(), table)
```

Comment:
So the untyped API is still reading as `TableRow`, is this preferred over using Avro here as well? More convenient but less performant?

Reply:
For backwards compatibility mostly. Generic Avro API is just as bad as JSON even if it might be slightly faster.

Reply:
👍